data_write/data_write.py

```diff
@@ -600,23 +789,33 @@ def write_bfiq_params(parameters_holder):
                 parameters['antenna_arrays_order'] = []
 
                 flattened_data = []
+                num_antenna_arrays = 0
                 if "main" in bfiq[slice_id]:
+                    num_antenna_arrays += 1 
                     parameters['antenna_arrays_order'].append("main")
                     flattened_data.append(bfiq[slice_id]['main']['data'])
                 if "intf" in bfiq[slice_id]:
+                    num_antenna_arrays += 1 
                     parameters['antenna_arrays_order'].append("intf")
                     flattened_data.append(bfiq[slice_id]['intf']['data'])
 
                 flattened_data = np.concatenate(flattened_data)
                 parameters['data'] = flattened_data
 
-                parameters['num_samps'] = np.uint32(bfiq[slice_id].pop('num_samps', None))
-                parameters['data_dimensions'] = np.array([len(bfiq[slice_id].keys()),
+                parameters['num_samps'] = np.uint32(bfiq[slice_id]['num_samps'])
+                parameters['data_dimensions'] = np.array([num_antenna_arrays,
```

Comment:
I assume the data dimensions array is listed in order of highest level dimension to lowest level? As in, the number of arrays is 0, 1 or 2; the number of integrations is ~30; the number of beams is 16, then the number of samples is large... do you think the number of beams and the number of integrations should be swapped in the array? What is 'higher level'?

Reply:
we think it should remain as is. Number of integrations is a physical number transmitted, beams were a result of processing